internal/xds/resolver/xds_resolver.go

```diff
@@ -205,50 +200,35 @@ func (*xdsResolverBuilder) Scheme() string {
 
 // xdsResolver implements the resolver.Resolver interface.
 //
-// It registers a watcher for ServiceConfig updates with the xdsClient object
-// (which performs LDS/RDS queries for the same), and passes the received
-// updates to the ClientConn.
+// It manages the dependency manager which in turn manages all xDS resource
+// watches. It receives the xDS resource config and passes them to ClientConn.
 type xdsResolver struct {
-	cc     resolver.ClientConn
-	logger *grpclog.PrefixLogger
+	// The following fields are initialized at creation time and are read-only
+	// after that.
+	cc              resolver.ClientConn
+	logger          *grpclog.PrefixLogger
+	ldsResourceName string
+	dm              *xdsdepmgr.DependencyManager
 	// The underlying xdsClient which performs all xDS requests and responses.
```

Comment:
This comment was probably added way back when we were still getting used to what the xDS client was. I think we can remove it now. Doesn't add much value.